components/odh-notebook-controller/main.go

```diff
@@ -59,6 +60,17 @@ func init() {
 	//+kubebuilder:scaffold:scheme
 }
 
+func getControllerNamespace() (string, error) {
+	// Try to get the namespace from the service account secret
+	if data, err := os.ReadFile("/var/run/secrets/kubernetes.io/serviceaccount/namespace"); err == nil {
+		if ns := string(data); len(ns) > 0 {
+			return ns, nil
+		}
+	}
+
+	return "", fmt.Errorf("unable to determine the namespace")
```

Comment:
I would've probably used the previous default we had

```go
	// Fallback to default namespace, keep default as redhat-ods-applications
	return "redhat-ods-applications"
```

but that's just my ingrained conservatism speaking. There should not be anything wrong in Exit(1) in this situation.

I just wonder how our e2e tests are going to deal with this. Could maybe explain why they were failing?

Reply:
let's not default to "redhat-ods-applications"